src/lib/DepositApiClient.js

```diff
@@ -89,21 +89,33 @@ export class RDMDepositApiClient extends DepositApiClient {
   }
 
   async _createResponse(axiosRequest) {
+    let responseData = {};
+    let shouldThrowError = false;
+
     try {
       const response = await axiosRequest();
-      const data = this.recordSerializer.deserialize(response.data || {});
-      const errors = this.recordSerializer.deserializeErrors(
-        response.data.errors || []
-      );
-      return new DepositApiClientResponse(data, errors);
+      responseData = response.data;
     } catch (error) {
-      const data = this.recordSerializer.deserialize(error.response.data || {});
-      const errors = this.recordSerializer.deserializeErrors(
-        error.response.data.errors || []
-      );
-      const wrapped = new DepositApiClientResponse(data, errors);
-      throw new Error(wrapped);
+      shouldThrowError = true;
+      responseData = error.response.data;
     }
+
+    const data = this.recordSerializer.deserialize(responseData || {});
+
+    const noFormFieldErrors =
+      !responseData.hasOwnProperty('errors') && shouldThrowError;
+
+    const errors = noFormFieldErrors
```

Comment:
I am confused by this. If no form errors, the errors is the `responseData`?

General feeling: I have trouble understanding the code by quickly reading it, with multiple`if`s around? Wasn't it more readable before (even if there was some little duplication)?

Reply:
i agree that it was a bit too simplified, but i would argue that it is more descriptive than before now.

Reply:
I agree with @ntarocco , the initial suggestion was meant to improve readability. But I feel we traded duplication for conditional logic, especially ternary operators, which make the code a bit harder to read right now.

Most of the conditional logic might not be needed, since JS already does it for us when you write code like:

```javascript
const errors = this.recordSerializer.deserializeErrors(responseData.errors || [])
```

I'd argue only one `if` statement is needed for the whole function, but we can discuss it @NRodriguezcuellar.

Also, feel free to discard my suggestions if they do not improve readability at all.